cinema/views.py

```diff
@@ -21,24 +23,49 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly, )
+    authentication_classes = (TokenAuthentication,)
```

Comment:
this is a good approach, but if you use the same auth_classes or permissions_classes almost everywhere, you can set up default classes in REST_FRAMEWORK constant. Here are docs: https://www.django-rest-framework.org/api-guide/permissions/#setting-the-permission-policy https://www.django-rest-framework.org/api-guide/authentication/#setting-the-authentication-scheme
If some view requires different classes, you can just override view attributes(like you do it right now)